### ISSUE TITLE
NRQL syntax errors in newrelic_one_dashboard does not produce error in Terraform (run is successful)

### ISSUE THREAD

**Issue by Junkern** [newrelic/terraform-provider-newrelic#1318]
### Terraform Version
```
Terraform v0.15.1
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v3.42.0
+ provider registry.terraform.io/newrelic/newrelic v2.22.1
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `newrelic_one_dashboard`

### Terraform Configuration

```hcl
provider "newrelic" {
  api_key = "NRAK-xxxxxx"
  account_id = "xxxxx"
  region = "US"
}
```

### Actual Behavior

I have a `newrelic_one_dashboard` with two existing `page` blocks. I want to add a third and fourth page block. A `plan` correctly shows the page blocks to be added and executing the `apply` results in a (apparently) successful run: `Apply complete! Resources: 0 added, 1 changed, 0 destroyed.`

However, visiting the dashboard shows me, that the additional page blocks do not show up. Moreover, a `terraform plan` shows me, that terraform would like to create those blocks again.

So, the creation of the pages does not work but still shows as successful.

I have tried to create a reproducible case, but was not successful, yet.

I also noticed that sometimes, terraform (or the newrelic backend) seems to suddenly change GUIDs and, thus, forgets about an already existing dashboard. I have had it happen a few times, that the `plan` on a already existing dashboard wants to create a new dashboard (although the existing one was not deleted)

### Expected Behavior

Adding multiple page blocks should add them succesfully to a dashboard

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Debug Output
https://gist.github.com/Junkern/0ccf48eaf9e76390faccad566d0bc2ef

**Comment by Junkern** [comment]
I currently have two page blocks and wanted to add two more (which did not work). After reading https://github.com/newrelic/terraform-provider-newrelic/issues/1248 I commented out one of the two page blocks and now it did work.

I indeed find a syntax error and now `apply` works on both new page blocks.

I wonder, however, why the terraform-newrelic-provider did not throw any error about the NRQL syntax error...
